compiler/test/dotty/tools/repl/ShadowingTests.scala

```diff
@@ -33,7 +33,7 @@ object ShadowingTests:
     val subdir = dir.resolve(name)
     try Files.createDirectory(subdir)
     catch case _: java.nio.file.FileAlreadyExistsException =>
-    assert(Files.isDirectory(subdir), s"failed to create shadowed subdirectory $subdir")
+      assert(Files.isDirectory(subdir), s"failed to create shadowed subdirectory $subdir")
```

Comment:
OK, fixed. Goes to show that this really is a dangerous trap.

Can we make it an error? I don't see how without undermining fundamental assumptions how the parser works. According to the syntax we need an `Expr` and that's what we can parse. Rejecting it would mean we throw aspects which are outside the context-free syntax into the parsing algorithm.

Reply:
No wait, it's asserting that the existing thing is a dir, right? If createDir succeeds, it better be a dir.

Reply:
One way to clarify the syntax is to require the expr to begin on the same line as the arrow. So the arrow could move to the following line.

Reply:
In fact the warning occurrences in the backend were also assuming an empty 
catch handler. I changed them to do the right thing, and fixed the warning message
to point out this possibility.

Reply:
@som-snytt I think you are right for ShadowingTests. The assert really is meant as a handler. @griggt can you confirm?

Reply:
I'm almost certain that when I wrote it I meant for the catch handler to be empty.  But in this case it works either way.

Reply:
In fact, we need more forms of syntax that work either way. If Booleans worked either way, we'd never take the wrong branch.